### DISCUSSION TITLE
Support of custom column names instead of default `email` for login

### DISCUSSION THREAD

**Discussion by dragomano** [filamentphp/filament#12284] [Ideas]
I'm using custom members table where email stored in `email_address`, but your admin panel tries to get email from `email` column. How and where can this be customized?

**Comment by manassegitau** [comment]
> I'm using custom members table where email stored in `email_address`, but your admin panel tries to get email from `email` column. How and where can this be customized?

I am also experiencing this. Did you get a workaround for that?

**Comment by manassegitau** [comment]
> I'm using custom members table where email stored in `email_address`, but your admin panel tries to get email from `email` column. How and where can this be customized?

I found the solution to this. You can customize any of the auth pages by creating a custom class and then modifying the method that you need to. You can find this on the documentation [here](https://filamentphp.com/docs/3.x/panels/users#customizing-the-authentication-features).
In this case you need to modify the Login page so create `Login.php` in `Filament\Pages\Auth\` and modify the email column name to match yours
```
<?php
use Filament\Pages\Auth\Login as BaseLogin;

class Login extends BaseLogin
{
    protected function getCredentialsFromFormData(array $data): array
    {
        return [
            'user_email' => $data['email'],
            'password' => $data['password'],
        ];
    }
}
```
Another issue I have also realised one might encounter is when you have modified the password column name. If you have, you need to update the `User ` model as follows:
```
 public function getAuthPassword()
    {
        return $this->password_column_name;
    }

 public function getAuthPasswordName()
    {
        return 'password_column_name';
    }
``` 
Hope that helps.
